.github/workflows/check_data_leaks.yml

```diff
@@ -0,0 +1,21 @@
+name: Check Data Leaks
+
+on: [pull_request]
+
+defaults:
+  run:
+    working-directory: .
+
+jobs:
+  scan-data-leaks:
+    name: gitleaks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
```

Comment:
```suggestion
```

This is probably useless to clone the whole repository. Latest versions is probably enough.